primitives/src/assets.rs

```diff
@@ -444,7 +444,7 @@ where
 			<Native as Currency<C::AccountId>>::deposit_creating(beneficiary, amount);
 		} else {
 			<NonNative as Mutate<C::AccountId>>::mint_into(asset_id, beneficiary, amount)
-				.map_err(|e| FungibleLedgerError::InvalidMint(e))?;
+				.map_err(FungibleLedgerError::InvalidMint)?;
```

Comment:
> Recall from the [“Enum values”](https://doc.rust-lang.org/stable/book/ch06-01-defining-an-enum.html#enum-values) section of Chapter 6 that the name of each enum variant that we define also becomes an initializer function. We can use these initializer functions as function pointers that implement the closure traits, which means we can specify the initializer functions as arguments for methods that take closures

https://doc.rust-lang.org/stable/book/ch19-05-advanced-functions-and-closures.html#function-pointers

This might be the answer.